backoffice/workflows/api/views.py

```diff
@@ -98,6 +99,56 @@ def resolve(self, request, pk=None):
                 ResolutionDags[serializer.validated_data["value"]].label, pk, extra_data
             )
 
+    @action(detail=True, methods=["post"])
+    def restart(self, request, pk=None):
+
+        params = request.data.get("params", None)
+        restart_current_task = request.data.get("restart_current_task", False)
+
+        workflow = Workflow.objects.get(id=pk)
+
+        data = {"dry_run": False, "dag_run_id": pk, "reset_dag_runs": True}
+
+        executed_dags_for_workflow = {}
+        # find dags that were executed
+        for dag_id in AUTHOR_DAGS[workflow.workflow_type]:
+            response = requests.get(
+                f"{airflow_utils.AIRFLOW_BASE_URL}/api/v1/dags/{dag_id}/dagRuns/{pk}",
+                json=data,
+                headers=airflow_utils.AIRFLOW_HEADERS,
+            )
+            if response.status_code == status.HTTP_200_OK:
+                executed_dags_for_workflow[dag_id] = response.content
```

Comment:
why we need to iterate through all the dags? for example this code will run both `approve` and `reject` DAGS

Reply:
For that case it will simply check which one of them got executed and restart it. (if the response is 200 OK it means it got executed - regardless of finishing successfully on)

In general this code serves to fetch which dags of a given workflow have run. 
For example there can be cases that, for a given workflow, only the `author_create_initialization_dag` dag ran and no `approve` nor `reject` got triggered . in this case it will only restart this 1 dag